cmd/atlas/internal/ci/run.go

```diff
@@ -35,50 +36,80 @@ type Runner struct {
 
 	// ReportWriter writes the summary report.
 	ReportWriter ReportWriter
+
+	// summary report. reset on each run.
+	sum *SummaryReport
 }
 
 // Run executes the CI job.
 func (r *Runner) Run(ctx context.Context) error {
-	sum, err := r.summary(ctx)
-	switch err := err.(type) {
-	case nil:
-		return r.ReportWriter.WriteReport(sum)
-	case *FileError:
-		return r.ReportWriter.WriteReport(&SummaryReport{
-			Files: []*FileReport{{Name: err.File, Error: err.Error()}},
-		})
-	default:
-		return err
+	err := r.summary(ctx)
+	if err2 := r.ReportWriter.WriteReport(r.sum); err2 != nil {
+		if err != nil {
+			err2 = fmt.Errorf("%w: %v", err, err2)
+		}
+		err = err2
 	}
+	return err
 }
 
-func (r *Runner) summary(ctx context.Context) (*SummaryReport, error) {
-	// Validate sum file in case it exists.
-	if err := migrate.Validate(r.Dir); err != nil && !errors.Is(err, migrate.ErrChecksumNotFound) {
-		return nil, &FileError{File: migrate.HashFileName, Err: err}
+const (
+	stepIntegrityCheck = "Migration Integrity Check"
+	stepDetectChanges  = "Detect New Migration Files"
+	stepLoadChanges    = "Replay Migration Files"
+	stepAnalyzeFile    = "Analyze %s"
+)
+
+func (r *Runner) summary(ctx context.Context) error {
+	r.sum = NewSummaryReport(r.Dev, r.Dir)
+
+	// Integrity check.
+	switch err := migrate.Validate(r.Dir); {
+	case errors.Is(err, migrate.ErrChecksumNotFound):
```

Comment:
what happens if err is migrate.ErrChecksumNotFound?

Reply:
This step is skipped